<?xml version="1.0"?>
<launch>
  <param name="use_sim_time" value="true"/>
  <arg name="cmd_mode_ackermann" default="true" />
  <arg name="config_file" default="$(find simulator_art)/config/vehicle_model.yaml"/>
  
  <!-- Rviz -->
  <arg name="rviz" default="true" />
  <arg name="rviz_file" default="$(find simulator_art)/rviz/sim.rviz" />

  <rosparam command="load" file="$(arg config_file)" ns="artstage"/>
	
  <!-- stage simulator for vehicle -->
  <!-- launch-prefix="xterm -e gdb - -args" -->
  <node pkg="simulator_art" type="stageros" name="stageros" args="-u $(find simulator_art)/stage_worlds/scu.world" output="screen"  >
    <rosparam file="$(find simulator_art)/stage_worlds/scu.yaml"/>
    <param name="cmd_mode_ackermann" value="$(arg cmd_mode_ackermann)"/>
    <param name="broadcast_utm_odom" value="true"/>
    <param name="delay_location_steps" value="16"/>
  </node>
  
  <!-- teleop keyboard-->
  <include file="$(find ackermann_drive_teleop)/launch/ackermann_drive_keyop.launch"  >
    <arg name="command_topic" value="/ackermann_cmd"/>
    <arg name="max_speed" value="4.5"/>
    <arg name="max_angle" value="0.7"/>
    <arg name="xterm" value="true"/>
  </include>

  <!-- rviz visualization-->
  <param name="robot_description" textfile="$(find simulator_art)/rviz/vehicle.urdf"/>
  <node pkg="rviz" type="rviz" name="$(anon rviz)" args="-d $(arg rviz_file)" output="screen" if="$(arg rviz)"/>
  
</launch>

